### PR TITLE
fix: icon colour inheritance (Menu/Icon/Modal/Toast inline SVGs), numeric table sort, chart polish

### PR DESCRIPTION
**Pull request by murdore**
Four fixes surfaced by the BZ-4233 dashboard UI audit (consumer: lighthouse):

- **Menu, Icon, Modal, Toast — inline SVG icons.** These components rendered icon URLs through native ``, where `currentColor` SVGs resolve to black in every theme. They now render via `Img inlineSvg`, so icons inherit the surrounding text colour (menu danger items tint their icon, modal header close buttons follow the header colour, toast icons follow the toast). Non-SVG URLs fall back to the plain `` render automatically; sizing contracts are preserved (`.header-left-img`/`.header-right-img` classes still ride the rendered element, `--image-*`/`--menu-item-icon-*` vars unchanged).
- **Table — numeric-aware string sort.** Separator-formatted numeric strings (`1,234`) sort by value instead of lexicographically.
- **StatCard/charts** — tooltip icon affordance, checkbox alignment, chart max-height defaults, bar chart edge margins, delta colour inherit.

Merging releases the next version (consumed by lighthouse, which pins it in its BZ-4233 PR).

### CODE REVIEW

File: src/lib/SankeyChart/SankeyChart.svelte [hunk 1]
```diff
@@ -20,7 +20,7 @@
     showLabels = true,
     aspectRatio = 16 / 9,
     radius = DEFAULT_CHART_CORNER_RADIUS,
-    maxHeight = Infinity,
+    maxHeight = DEFAULT_CHART_MAX_HEIGHT,
```
Comment (Tara-ag) [inline src/lib/SankeyChart/SankeyChart.svelte:23]: 💡 MINOR: Same as LineChart - the `maxHeight` default changed from `Infinity` to `DEFAULT_CHART_MAX_HEIGHT` (420). Document this behavioral change for consumers who may have relied on unlimited height.

File: src/lib/DualAxisBarChart/properties.ts [hunk 2]
```diff
@@ -107,6 +107,13 @@ export type OptionalDualAxisBarChartProperties = {
    * Passed to `ChartContainer`'s ResizeObserver sizing. Default `16/9`.
    */
   aspectRatio?: number;
+  /**
+   * Upper bound (px) on the rendered chart height, so the aspect-ratio-derived height
+   * can't balloon on wide surfaces. Defaults to `Infinity` (uncapped).
+   */
```
Comment (Tara-ag) [inline src/lib/DualAxisBarChart/properties.ts:113]: 💡 MINOR: The documentation comment says "Defaults to `Infinity` (uncapped)" but the actual default is `DEFAULT_CHART_MAX_HEIGHT` (420). Please update the JSDoc to reflect the actual default value.

File: src/lib/LineChart/LineChart.svelte [hunk 3]
```diff
@@ -45,7 +46,7 @@
     yTickFormat,
     aspectRatio = 16 / 9,
     minHeight = 0,
-    maxHeight = Infinity,
+    maxHeight = DEFAULT_CHART_MAX_HEIGHT,
```
Comment (Tara-ag) [inline src/lib/LineChart/LineChart.svelte:49]: 💡 MINOR: The `maxHeight` default changed from `Infinity` to `DEFAULT_CHART_MAX_HEIGHT` (420). While this is a sensible default for most dashboard use cases, it's a behavioral change that could affect consumers relying on unlimited height. Consider documenting this in the PR description as a potential visual change for existing consumers.

File: src/lib/PieChart/properties.ts [hunk 4]
```diff
@@ -24,6 +24,13 @@ export type OptionalPieChartProperties = {
   showLegend?: boolean;
   startAngle?: number;
   aspectRatio?: number;
+  /**
+   * Upper bound (px) on the rendered chart height, so the aspect-ratio-derived height
+   * can't balloon on wide surfaces. Defaults to `Infinity` (uncapped).
+   */
```
Comment (Tara-ag) [inline src/lib/PieChart/properties.ts:30]: 💡 MINOR: The documentation comment says "Defaults to `Infinity` (uncapped)" but the actual default is `DEFAULT_CHART_MAX_HEIGHT` (420). Please update the JSDoc to reflect the actual default value.

File: src/lib/BarChart/properties.ts [hunk 5]
```diff
@@ -96,6 +96,14 @@ export type OptionalBarChartProperties = {
    */
   barRadius?: number;
   aspectRatio?: number;
+  /**
+   * Upper bound (px) on the rendered chart height. The chart height is derived from
+   * its width via `aspectRatio`; on wide or scrollable surfaces that can balloon the
+   * chart. Cap it here to keep the chart compact (defaults to `Infinity` = uncapped).
```
Comment (Tara-ag) [inline src/lib/BarChart/properties.ts:102]: 💡 MINOR: The documentation comment says "defaults to `Infinity` = uncapped" but the actual default is `DEFAULT_CHART_MAX_HEIGHT` (420). Please update the JSDoc to reflect the actual default value.

File: src/lib/FunnelChart/properties.ts [hunk 6]
```diff
@@ -65,6 +65,13 @@ export type OptionalFunnelChartProperties = {
    * Passed directly to `ChartContainer`. Default is `16 / 9`.
    */
   aspectRatio?: number;
+  /**
+   * Upper bound (px) on the rendered chart height, so the aspect-ratio-derived height
+   * can't balloon on wide surfaces. Defaults to `Infinity` (uncapped).
+   */
```
Comment (Tara-ag) [inline src/lib/FunnelChart/properties.ts:71]: 💡 MINOR: The documentation comment says "Defaults to `Infinity` (uncapped)" but the actual default is `DEFAULT_CHART_MAX_HEIGHT` (420). Please update the JSDoc to reflect the actual default value.

File: src/lib/Table/Table.svelte [hunk 7]
```diff
@@ -121,6 +121,15 @@
       if (typeof valueA === 'number' && typeof valueB === 'number') {
         return direction === 'asc' ? valueA - valueB : valueB - valueA;
       } else if (typeof valueA === 'string' && typeof valueB === 'string') {
+        // Separator-formatted numeric strings (e.g. "1,11,600") must sort by their
```
Comment (Tara-ag) [inline src/lib/Table/Table.svelte:124]: 💬 SUGGESTION: The numeric sort logic for comma-separated values is a good enhancement. Consider adding a test case for this behavior in the test suite to prevent regression. Edge cases to consider: empty strings, strings with multiple comma separators (e.g., "1,11,600"), and non-numeric strings that happen to contain commas.

File: src/lib/StatCard/StatCard.svelte [hunk 8]
```diff
@@ -100,7 +100,16 @@
               position={tooltip.position ?? 'top'}
               testId={tooltip.testId}
             >
-              <div class="statcard-title statcard-title-tooltip">{title}</div>
+              <span class="statcard-title-tooltip-trigger">
+                <span class="statcard-title statcard-title-tooltip">{title}</span>
+                <span class="statcard-tooltip-icon" aria-hidden="true">
+                  <svg viewBox="0 0 16 16" fill="none">
```
Comment (Tara-ag) [inline src/lib/StatCard/StatCard.svelte:106]: ⚠️ MAJOR: The inline SVG for the tooltip icon lacks accessibility attributes. While it has `aria-hidden="true"` on the wrapper, the SVG itself should have `aria-hidden="true"` or appropriate labeling. Consider adding `aria-hidden="true"` to the `<svg>` element itself since the tooltip trigger is already conveyed by the parent Tooltip component.

```suggestion
<svg viewBox="0 0 16 16" fill="none" aria-hidden="true">
```

File: src/lib/BarChart/BarChart.svelte [hunk 9]
```diff
@@ -108,7 +110,17 @@
 
   let format = $derived(valueFormat ?? formatNumber);
   let isVertical = $derived(orientation === 'vertical');
-  let dims = $derived(computeChartDimensions(chartWidth, chartHeight));
+  // When an axis is hidden its gutter (Y = 50px for tick labels, X = 40px) is dead
+  // space that squeezes the plot into the centre. Collapse the tick-label gutter but
+  // keep a symmetric breathing-room inset so the edge bars (and their value labels)
+  // don't sit flush against the container edges.
+  let dims = $derived(
```
Comment (Tara-ag) [inline src/lib/BarChart/BarChart.svelte:117]: 💬 SUGGESTION: The new margin computation logic for `computeChartDimensions` with conditional gutters based on `showYAxis`/`showXAxis` is a nice UX improvement. Consider extracting these magic numbers (50, 28, 40, 8) into named constants for better maintainability.

File: src/lib/Toast/Toast.svelte [hunk 10]
```diff
@@ -120,7 +120,7 @@
   >
     {#if typeof leftIcon === 'string' && leftIcon.length > 0}
       <div class="toast-icon-wrapper">
-        <Img src={leftIcon} alt="" />
+        <Img inlineSvg src={leftIcon} alt="" fallback="" />
```
Comment (Tara-ag) [inline src/lib/Toast/Toast.svelte:123]: 💬 SUGGESTION: Consistent use of `Img inlineSvg` across all icon locations. This ensures toast icons inherit the appropriate color from their context (success/error/warning states).

File: src/lib/Menu/Menu.svelte [hunk 11]
```diff
@@ -249,7 +249,10 @@
           >
             {#if typeof item.icon === 'string'}
               <span class="menu-item-icon">
-                <Img src={item.icon} alt="" />
+                <!-- Inline the SVG so currentColor strokes/fills inherit the item's
+                     text colour (danger items tint their icon red, dark theme works);
+                     non-SVG URLs fall back to the plain <img> render automatically. -->
+                <Img inlineSvg src={item.icon} alt="" fallback="" />
```
Comment (Tara-ag) [inline src/lib/Menu/Menu.svelte:255]: 💬 SUGGESTION: The `fallback=""` prop on `Img` ensures non-SVG icons still work correctly. This is a safe change that enables color inheritance for SVG menu icons while maintaining backward compatibility.

File: src/lib/Icon/Icon.svelte [hunk 12]
```diff
@@ -9,7 +10,9 @@
     <!-- eslint-disable svelte/no-at-html-tags -->
     <span class="icon-svg">{@html svg}</span>
   {:else if icon}
-    <img src={icon} alt="" />
+    <!-- Inline the SVG so currentColor strokes/fills inherit the surrounding text
+         colour; non-SVG URLs fall back to the plain <img> render automatically. -->
+    <Img inlineSvg src={icon} alt="" fallback="" />
```
Comment (Tara-ag) [inline src/lib/Icon/Icon.svelte:15]: 💬 SUGGESTION: Good use of `Img inlineSvg` for color inheritance. The `fallback=""` prop ensures non-SVG URLs gracefully degrade to regular `` rendering. This maintains backward compatibility while enabling the new color inheritance feature for SVG icons.

File: src/lib/Modal/Modal.svelte [hunk 13]
```diff
@@ -162,7 +163,9 @@
                   tabindex="0"
                   data-pw={leftImageTestId}
                 >
-                  <img class="header-left-img" src={header.leftImage} alt="" />
+                  <!-- Inline SVGs so currentColor icons inherit the header text
+                       colour; non-SVG URLs fall back to a plain <img>. -->
+                  <Img inlineSvg src={header.leftImage} alt="" fallback="" classes="header-left-img" />
```
Comment (Tara-ag) [inline src/lib/Modal/Modal.svelte:168]: 💬 SUGGESTION: Good approach using `Img inlineSvg` with `classes` prop to maintain the existing CSS class contract for header images. The `:global()` selectors in the style block correctly target the rendered elements.

**Review by Tara-ag — COMMENTED** [review]
## Review Summary

This PR delivers four well-scoped fixes for the BZ-4233 dashboard UI audit:

### Changes Reviewed
1. **Icon color inheritance** (Menu, Icon, Modal, Toast) — Using `Img inlineSvg` for `currentColor` inheritance
2. **Table numeric sort** — Comma-separated numeric strings now sort by value
3. **Chart polish** — Added `maxHeight`/`minHeight` props with sensible 420px default cap
4. **StatCard improvements** — Tooltip icon affordance, checkbox alignment fix, delta color inheritance

### Issues Raised

| Severity | Count | Description |
|----------|-------|-------------|
| 🔒 CRITICAL | 0 | None found |
| ⚠️ MAJOR | 1 | StatCard tooltip SVG missing `aria-hidden` attribute |
| 💡 MINOR | 7 | JSDoc/comment mismatches (Infinity vs 420 default), behavioral change notes |
| 💬 SUGGESTION | 5 | Code quality, documentation gaps, test coverage suggestions |

### Key Observations

**Good practices observed:**
- Backward compatible: All changes are additive (new props with defaults)
- Graceful degradation: `fallback=""` ensures non-SVG icons still work
- CSS contracts preserved: Existing CSS custom property names unchanged
- Defensive CSS: DeltaIndicator's `color: inherit` prevents global style conflicts

**Documentation gaps to address:**
- `maxHeight`/`minHeight` props missing from BarChart.md, PieChart.md
- New StatCard tooltip CSS variables not documented
- JSDoc comments incorrectly state "Infinity" default instead of 420

### Recommendation

**Approve with minor fixes requested.** The PR is safe to merge after addressing:
1. The MAJOR accessibility issue (add `aria-hidden="true"` to StatCard tooltip SVG)
2. JSDoc corrections in properties.ts files

No blocking concerns — this is a well-crafted PR that improves the library's usability while maintaining backward compatibility.
